webpack/InsightsCloudSync/Components/InsightsSettings/InsightsSettingsReducer.js

```diff
@@ -0,0 +1,40 @@
+import Immutable from 'seamless-immutable';
+import {
+  INSIGHTS_SYNC_SETTING_SET,
+  INSIGHTS_SYNC_SETTING_SET_ERROR,
+  INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
+  INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
+} from './InsightsSettingsConstants';
+
+const initialState = Immutable({
+  insightsSyncEnabled: false,
+});
+
+export default (state = initialState, action) => {
+  const { payload: { settings: insightsSyncEnabled, error } = {} } = action;
+
+  switch (action.type) {
+    case INSIGHTS_SYNC_SETTINGS_GET_SUCCESS:
+      return state.merge({
+        ...state,
+        insightsSyncEnabled,
```

Comment:
for future additions, we can destruct the settings object here:

```suggestion
       return state.merge({
￼        ...state,
￼        ...settings,
       })
```